packages/core/src/types.ts

```diff
@@ -1005,23 +1005,26 @@ export type ContextFactory<TContext extends MachineContext> = ({
   input: any; // TODO: fix
 }) => TContext;
 
-export interface MachineConfig<
+export type MachineConfig<
   TContext extends MachineContext,
   TEvent extends EventObject,
   TAction extends ParameterizedObject = ParameterizedObject,
   TActorMap extends ActorMap = ActorMap,
   TTypesMeta = TypegenDisabled
-> extends StateNodeConfig<NoInfer<TContext>, NoInfer<TEvent>, TAction> {
+> = (StateNodeConfig<NoInfer<TContext>, NoInfer<TEvent>, TAction> & {
   /**
    * The initial context (extended state)
    */
-  context?: InitialContext<LowInfer<TContext>>;
   /**
    * The machine's own version.
    */
   version?: string;
   types?: MachineTypes<TContext, TEvent, TActorMap, TTypesMeta>;
-}
+  getContext?: (ctx: TContext) => void;
+}) &
+  (Equals<TContext, MachineContext> extends true
```

Comment:
This probably needs to be tweaked @Andarist - I'm comparing to `MachineContext` since it's the default type if none other is specified. Maybe we should allow `never` here?

Reply:
I'm a little bit unsure what kind of effect this intersection might have on inference, gonna have to recheck this. Other than that, this looks pretty OK.